test/test_time.cpp

```diff
@@ -50,6 +50,74 @@ class TestTimeFixture : public ::testing::Test
   }
 };
 
+// Tests the rcutils time unit conversion macros.
+TEST_F(TestTimeFixture, test_rcutils_time_conversion_macros) {
+  int64_t ns = 0;
+
+  // Note: 9007199254740993 or higher cannot be represented anymore by intermediate type double
+  // without a loss of precision.
+
+  // seconds to nanoseconds
+  EXPECT_EQ(ns = RCUTILS_S_TO_NS(1), 1000000000ll);  // int
+  EXPECT_EQ(ns = RCUTILS_S_TO_NS(0.2), 200000000ll);  // double
```

Comment:
CI for windows is showing a lot of new compiler warnings (on this line and others).  This is because MSVC is very pedantic about types, so assigning a double to an int64_t is a possible loss of precision.  Can you fix these types up (you can see more in the test output at https://ci.ros2.org/job/ci_windows/11192/msbuild/new/)?

Also, out of curiousity, why are you doing the assignment to `ns` at all here?

Reply:
Exactly for that, to make the implicit conversion to `int64_t` explicit. Otherwise the result of `RCUTILS_S_TO_NS(0.2)` would be of type double and `EXPECT_EQ()` would probably also convert the right argument to double for the comparison.

So the "best" and most explicit variant would be
```suggestion
  EXPECT_EQ(static_cast<int64_t>(RCUTILS_S_TO_NS(0.2)), 200000000ll);  // double
```